framework/include/utils/MooseUtils.h

```diff
@@ -383,6 +385,13 @@ tokenizeAndConvert(const std::string & str,
   }
   return true;
 }
+
+/**
+ * Convert supplied string to upper case.
+ * @params name The string to convert upper case, this is pass by value because the conversion
+ *              to upper case must modify the string.
+ */
+std::string toUpper(std::string name);
```

Comment:
Since you are returning a copy (which is a good design by the way), why don't you just modify the temporary you need instead of making a copy of the argument?